### PR TITLE
Docs/diataxis 1

### PR DESCRIPTION
**Pull request by aleksei-korolev**
# Description of change

Rewriting articles to get them in line with the style guide.

`documentation/docs/guide/core_concepts/iscp.md` is not used anywhere so I deleted it.

## Type of change

- Documentation Fix

## How the change has been tested

Describe the tests that you ran to verify your changes.

Make sure to provide instructions for the maintainer as well as any relevant configurations.

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [X] I have followed the [contribution guidelines](https://wiki.iota.org/smart-contracts/contribute) for this project
- [X] I have performed a self-review of my own code
- [X] I have selected the `develop` branch as the target branch
- [X] I have made corresponding changes to the documentation
- [X] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by aleksei-korolev** [comment]
@iotaledger/tech-writers

File: documentation/docs/guide/core_concepts/smart-contracts.md [hunk 1]
```diff
@@ -1,26 +1,27 @@
...
 # Smart Contracts
 
-## What Are Smart Contracts?
+Smart contracts are software applications that run on a distributed network with multiple validators executing and validating the same code. This prevents tampering with the execution of the program and ensures that the application behaves as expected. 
 
-Smart contracts are software applications that run on a distributed network with multiple validators executing and validating the same code. This ensures the application behaves as expected, and that there is no tampering in the execution of the program. 
+## Applications You Can Trust
 
-### Applications You Can Trust
+As you can be certain that the code that is being executed will never change, this results in applications you can trust. This allows you to use smart contracts for applications where there was a trust issue. The rules of the smart contract define what the contract can and can not do, making it a decentralized and a predictable decision maker.
```
Comment (charlesthompson3) [inline documentation/docs/guide/core_concepts/smart-contracts.md:17]:
```suggestion
As certain as you can be that the executed code will never change, this results in applications you can trust. This allows you to use smart contracts for applications where there was a trust issue. The rules of the smart contract define what the contract can and can not do, making it a decentralized and a predictable decision maker.
```

File: documentation/docs/guide/core_concepts/smart-contracts.md [hunk 2]
```diff
@@ -1,26 +1,27 @@
...
 
-As you can be certain that the code being executed is always the same (and will not change), this results in applications you can trust. This allows you to use smart contracts for applications where there was a trust issue. The rules of the smart contract define what the contract can and can not do, making it a decentralized and a predictable decision-maker.
+Smart contracts are used for all kinds of purposes. A recurring reason to use a smart contract is to automate certain actions without needing a centralized entity to enforce this specific action. For example, a smart contract could exchange a certain amount of IOTA tokens for a certain amount of land ownership. That smart contract would accept both the IOTA tokens and the land ownership and predictably exchange them between both parties. It excludes the risk of one of the parties not delivering on their promise: with a smart contract, code is law.
```
Comment (charlesthompson3) [inline documentation/docs/guide/core_concepts/smart-contracts.md]:
```suggestion
Smart contracts are used for all kinds of purposes. A recurring reason to use a smart contract for automating certain actions without needing a centralized entity to enforce this specific action. For example, a smart contract could exchange a certain amount of IOTA tokens for a certain amount of land ownership. That smart contract would accept both the IOTA tokens and the land ownership and predictably exchange them between both parties. It excludes the risk of one of the parties not delivering on their promise: with a smart contract, code is law.
```

File: documentation/docs/guide/core_concepts/smart-contracts.md [hunk 3]
```diff
@@ -1,26 +1,27 @@
...
 
-On a public blockchain, anyone willing to pay the fees for deploying a smart contract can deploy one. Once your smart contract has been deployed to the chain you no longer have the option to change it, and you can be assured that your smart contract application will be there as long as that blockchain exists. Smart contracts can communicate with one another, and you can invoke programmed public functions on a smart contract to trigger actions on a smart contract, or address the state of a smart contract.
+Smart contracts do not run on just a single computer. Instead, each validator of the blockchain has to run it, compare the results with others and synchronize the state of the network. They need to wait while their messages are carried over the Internet, which introduces delays that you cannot solve with stronger computers or quicker software, and this issue will only get worse if you try to add more validators to the network. With enough requests, any blockchain network will get congested, and its execution fees will ramp up.
```
Comment (charlesthompson3) [inline documentation/docs/guide/core_concepts/smart-contracts.md]:
```suggestion
Smart contracts do not run on just a single computer. Instead, each validator of the blockchain has to run it, compare the results with others, and synchronize the state of the network. They need to wait while their messages are carried over the Internet, which introduces delays that you cannot solve with stronger computers or quicker software, and this issue will only get worse if you try to add more validators to the network. With enough requests, any blockchain network will get congested, and its execution fees will ramp up.
```

**Review by charlesthompson3 — APPROVED** [review]
Small feedback/suggestions👌

File: documentation/docs/guide/core_concepts/smart-contracts.md (hunk 1 above)
Comment (lucas-tortora) [reply documentation/docs/guide/core_concepts/smart-contracts.md:17]: @charlesthompson3 I think your edit would change the paragraph's meaning.

File: documentation/docs/guide/core_concepts/smart-contracts.md (hunk 2 above)
Comment (lucas-tortora) [reply documentation/docs/guide/core_concepts/smart-contracts.md]: I think "is for" would work better in this case. wdyt?

File: documentation/docs/guide/core_concepts/smart-contracts.md [hunk 4]
```diff
@@ -1,26 +1,27 @@
...
 
-Smart contracts are used for all kinds of purposes. A recurring reason to use a smart contract is to automate certain actions without needing a centralized entity to enforce this specific action. A good example of this is a smart contract that can exchange a certain amount of IOTA tokens for a certain amount of land ownership. The smart contract will accept both the IOTA tokens and the land ownership, and will predictably exchange them between both parties without the risk of one of the parties not delivering on their promise. **With a smart contract, code is law**.
+## Scalable Smart Contracts
 
-### Scalable Smart Contracts
+Anyone could deploy a smart contract to a public smart contract chain. Once it is deployed, nobody would be able to change or delete it. Smart contracts can communicate with one another, and you can invoke programmed public functions on a smart contract to trigger its execution or check its state.
```
Comment (lucas-tortora) [inline documentation/docs/guide/core_concepts/smart-contracts.md]:
```suggestion
Anyone could deploy a smart contract to a public smart contract chain. Once it is deployed, nobody will be able to change or delete it. Smart contracts can communicate with one another, and you can invoke programmed public functions on a smart contract to trigger its execution or check its state.
```

File: documentation/docs/guide/core_concepts/isc-architecture.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+- Layer 2
+- Layer 1
+- explanation
+---
+
+# ISC Architecture
+
+With IOTA Smart Contracts, anyone can start their own chain and invite others to validate it. Together, these chains are called Layer 2.
+
+Each chain has its own state and smart contracts that cause this state to change. As validators execute smart contracts, they tally these changes and write them into the chain. Each time they update the state, they collectively agree on a new state and commit to it by publishing its hash to Layer 1, the Tangle.
```
Comment (lucas-tortora) [inline documentation/docs/guide/core_concepts/isc-architecture.md:16]: "that cause this state to change" sounds like just deploying the contract could change the state. wdyt? 
```suggestion
Each chain has its state and smart contracts that can change state. As validators execute smart contracts, they tally these changes and write them into the chain. Each time they update the state, they collectively agree on a new state and commit to it by publishing its hash to Layer 1, the Tangle.
```

File: documentation/docs/guide/core_concepts/isc-architecture.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+Each Layer 2 chain is functionally equivalent to, say, Ethereum, but all chains can talk to Layer 1 and each other, making ISC a more intricate protocol.
```
Comment (lucas-tortora) [inline documentation/docs/guide/core_concepts/isc-architecture.md]:
```suggestion
Each Layer 2 chain is functionally equivalent to Ethereum, for example, but all chains can communicate with Layer 1 and each other, making ISC a more intricate protocol.
```

File: documentation/docs/guide/core_concepts/smart-contracts.md (hunk 3 above)
Comment (lucas-tortora) [reply documentation/docs/guide/core_concepts/smart-contracts.md]:
```suggestion
Smart contracts do not run on just a single computer. Instead, each validator of the blockchain has to run it, compare the results with others and synchronize the state of the network. They need to wait while their messages are carried over the Internet, introducing delays that you cannot solve with more powerful computers or quicker software. This issue will only get worse if you try to add more validators to the network. Any blockchain network will get congested with enough requests, and its execution fees will ramp up.
```

File: documentation/docs/guide/core_concepts/smart-contracts.md [hunk 7]
```diff
@@ -1,26 +1,27 @@
...
 
-Because smart contracts do not run on a single computer, but on many validators, a network of smart contracts can only process so many smart contracts at once, even if the software has been written optimally. This means smart contracts are expensive to execute, and do not scale well on a single blockchain, often resulting in congested networks and expensive fees for executing functions on smart contracts. **By allowing many blockchains executing smart contracts to run in parallel**, and communicate with one another, **IOTA Smart Contracts will solve this scalability problem.**
+As IOTA Smart Contracts run many independent chains, it spreads out the load and creates a network of much higher scale. At the same time, it provides advanced means of communication between its chains and preserves the ability to create complex, composed smart contracts.
```
Comment (lucas-tortora) [inline documentation/docs/guide/core_concepts/smart-contracts.md]:
```suggestion
As IOTA Smart Contracts run many independent chains, it spreads out the load and creates a network of a much larger scale. At the same time, it provides advanced means of communication between its chains and preserves the ability to create complex, composed smart contracts.
```

File: documentation/docs/overview.md [hunk 8]
```diff
@@ -1,40 +1,42 @@
...
-Allowing multiple blockchains to anchor to the tangle will solve several problems with smart contracts.
+## ISC Advantages
 
-## Scaling and Fees
+### Scaling and Fees
 
 Due to the ordered structure and execution time of a smart contract, a single blockchain can only handle so many transactions per second before it needs to decide on which transactions it needs to postpone until other blocks are produced, as there is no processing power available for them on that chain. This eventually results in high fees on many chains, and limited functionality. 
 
-As we allow many chains (which can communicate with other chains) to be anchored to the IOTA tangle, we can simply add additional chains once this becomes a problem. This results in lower fees over solutions that just use a single blockchain for all their smart contracts. 
+As ISC allows many chains to be anchored to the IOTA Tangle and lets them communicate with each other, we can simply add additional chains once this becomes a problem. This results in lower fees over solutions that just use a single blockchain for all their smart contracts.
```
Comment (lucas-tortora) [inline documentation/docs/overview.md]:
```suggestion
As ISC allows many chains to be anchored to the IOTA Tangle and lets them communicate, we can add additional chains once this becomes a problem. This results in lower fees over solutions that use a single blockchain for all their smart contracts.
```

File: documentation/docs/overview.md [hunk 9]
```diff
@@ -1,40 +1,42 @@
...
 
-## Custom Chains
+### Custom Chains
 
-Given that anyone can start a new chain, and set the rules for that chain, a lot is possible. Not only do you have full control over how the fees are handled on the chain you set up, but you also have full control over the validators and access to your chain. You can even spin up a private blockchain with no public data besides the state hash that is anchored to the main IOTA tangle. This allows parties that need private blockchains to use the security of the public IOTA network without actually exposing their blockchain to the public.
+Given that anyone can start a new chain and set its rules, a lot is possible. Not only do you have full control over how the fees are handled on the chain you set up, but you also have full control over the validators and access to your chain. You can even spin up a private blockchain with no public data besides the state hash that is anchored to the main IOTA Tangle. This allows parties that need private blockchains to use the security of the public IOTA network without actually exposing their blockchain to the public.
```
Comment (lucas-tortora) [inline documentation/docs/overview.md:26]:
```suggestion
Given that anyone can start a new chain and set its rules, a lot is possible. You have complete control over how the fees are handled on the chain you set up, but you also have full control over the validators and access to your chain. You can even spin up a private blockchain with no public data besides the state hash anchored to the main IOTA Tangle. This allows parties that need private blockchains to use the security of the public IOTA network without actually exposing their blockchain to the public.
```

**Review by lucas-tortora — APPROVED** [review]
Just a few small comments, LGTME :)

File: documentation/docs/guide/core_concepts/smart-contracts.md (hunk 1 above)
Comment (charlesthompson3) [reply documentation/docs/guide/core_concepts/smart-contracts.md:17]: I wrestled with it tbh. It feels disjointed to me but I could be wrong 🤷

File: documentation/docs/guide/core_concepts/smart-contracts.md (hunk 2 above)
Comment (charlesthompson3) [reply documentation/docs/guide/core_concepts/smart-contracts.md]: Yeah thats better

File: documentation/docs/guide/core_concepts/isc-architecture.md (hunk 5 above)
Comment (aleksei-korolev) [reply documentation/docs/guide/core_concepts/isc-architecture.md:16]: We didn't mention deployment yet, so I think this ambiguity does not exist for the reader yet, and by the point the reader will learn about SC deployment, he'll know about how the state mutations work already.

Btw, deployment of the SC does cause the blockchain state to change ;)

File: documentation/docs/guide/core_concepts/smart-contracts.md (hunk 2 above)
Comment (aleksei-korolev) [reply documentation/docs/guide/core_concepts/smart-contracts.md]: I've rewritten this sentence a bit.

File: documentation/docs/guide/core_concepts/smart-contracts.md (hunk 1 above)
Comment (aleksei-korolev) [reply documentation/docs/guide/core_concepts/smart-contracts.md:17]: The slight difference is that "executed code" has a connotation of "the code that is being executed at this exact moment", so "executed code will never change" has a meaning of "you cannot inject an instruction as the code is running". "The code that is being executed will never change" also means that if you order the network to run the code, this code will not change between the request and start of the execution, so you can inspect it before you decide to make this order.

Comment (aleksei-korolev) [reply documentation/docs/guide/core_concepts/smart-contracts.md:17]: Case 1: You order a big mac, as order is processed, the McGuy decides to not put lettuce into your burger.
Case 2: You order a big mac, but immediately after you pay they get an urgent message from the head office which says that big macs come without lettuce from that point on.

**Review by jorgemmsilva — APPROVED** [review]

**Comment by aleksei-korolev** [comment]
@jorgemmsilva I've rebased this one. Also I fixed the links that I broke in a different PR.

**Pull request merged**